app/services/mailchimp_subscription.rb

```diff
@@ -0,0 +1,48 @@
+class MailchimpSubscription
+
+  # Creates a member on the mailchimp mailing list with the given options
+  #
+  # @param [Hash] options The attributes of the mailing list member
+  # @option options [String] :email The member's email
+  # @option options [String] :username The member's username
+  # @option options [String] :signup_date The date at which the member signed up or registered
+  # @return [Gibbon::Response] The response from the mailchimp API
+  def self.create(options)
```

Comment:
This method is doing a bit too much, to make it more readable and changeable can you refactor this class to look something like this please?

```
class MailchimpSubscription
  attr_reader :email, :username, :signup_date
  private :email, :username, :signup_date
  
  def initialize(options)
    @email = options[:email]
    @username = options[:username]
    @signup_date = options[:signup_date]
  end
  
  def self.create(options)
    new(options).create
  end
  
  def create
    mailchimp_list.members.create(subscriber_details)
  end

  private
  
  def mailchimp_list
    request.lists(ENV['MAILCHIMP_LIST_ID'])
  end
  
   def subscriber_details
    {
      email_address: email,
      status: :subscribed,
      merge_fields: merge_fields
    }
  end
  
  def merge_fields
    {
      USERNAME: username,
      SIGNUPDATE: signup_date
    }
  end

  def request
    Gibbon::Request.new(api_key: ENV['MAILCHIMP_API_KEY'])
  end
end
```